src/mesh/mesh-amr_loadbalance.cpp

```diff
@@ -558,35 +546,32 @@ void Mesh::UpdateMeshBlockTree(int &nnew, int &ndel) {
 #ifdef MPI_PARALLEL
   if (tnref > 0) {
     PARTHENON_MPI_CHECK(MPI_Allgatherv(MPI_IN_PLACE, bnref[Globals::my_rank], MPI_BYTE,
-                                       lref, bnref.data(), brdisp.data(), MPI_BYTE,
+                                       lref.data(), bnref.data(), brdisp.data(), MPI_BYTE,
                                        MPI_COMM_WORLD));
   }
   if (tnderef >= nleaf) {
     PARTHENON_MPI_CHECK(MPI_Allgatherv(MPI_IN_PLACE, bnderef[Globals::my_rank], MPI_BYTE,
-                                       lderef, bnderef.data(), bddisp.data(), MPI_BYTE,
-                                       MPI_COMM_WORLD));
+                                       lderef.data(), bnderef.data(), bddisp.data(),
+                                       MPI_BYTE, MPI_COMM_WORLD));
   }
 #endif
 
   // calculate the list of the newly derefined blocks
   int ctnd = 0;
   if (tnderef >= nleaf) {
-    int lk = 0, lj = 0;
-    if (!mesh_size.symmetry(X2DIR)) lj = 1;
-    if (!mesh_size.symmetry(X3DIR)) lk = 1;
+    int lk = !mesh_size.symmetry(X3DIR);
+    int lj = !mesh_size.symmetry(X2DIR);
     for (int n = 0; n < tnderef; n++) {
-      if ((lderef[n].lx1() & 1LL) == 0LL && (lderef[n].lx2() & 1LL) == 0LL &&
-          (lderef[n].lx3() & 1LL) == 0LL) {
+      if (lderef[n].IsLowerLeftCornerOfParent()) {
         int r = n, rr = 0;
-        for (std::int64_t k = 0; k <= lk; k++) {
-          for (std::int64_t j = 0; j <= lj; j++) {
-            for (std::int64_t i = 0; i <= 1; i++) {
+        // offsets from "bottom left" corner of parent block
+        for (std::int64_t ok = 0; ok <= lk; ok++) {
+          for (std::int64_t oj = 0; oj <= lj; oj++) {
+            for (std::int64_t oi = 0; oi <= 1; oi++) {
```

Comment:
I cannot say that I have fully internalized the updated mesh structure yet, but double checking for my own understanding:
Why are `lk` and `lj` tied to `symmetry` but `i` is not?
I see that our grids are always at least 1D (which I'd understand overall a different behavior), but what determines the use of `symmetry`?
IIRC I had a similar question in the past and it was tied to being periodic but I'm not that sure any more.

Reply:
@lroberts36 should jump in but... as I understand it, `symmetry` is totally synonymous with dimension. It's essentially, "are we plane-symmetric in this direction?"

Reply:
Yes, `symmetry(dir)` being true means that we have translational symmetry in direction `dir`. Previously in many places in the code we just checked if the size of a block was one in a given direction, but this created some issues with multigrid when multiple block sizes were allowed. I think we could also probably remove the `symmetry` stuff and replace it with `ndim` checks.

Reply:
I prefer checking `symmetry`. It's more self-descriptive.

Reply:
Maybe it should be changed to `TranslationalSymmetry` or something because this isn't the first time someone has been confused about what that call implies.

Reply:
A reasonable suggestion for a later MR.

Reply:
yeah, when I wrote "Maybe it should be changed..." I was thinking the change should occur at some (possibly infinite) time in the future